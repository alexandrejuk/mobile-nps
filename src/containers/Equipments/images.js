import A630 from '../../assets/images/equipments/a6-30.png'
import A860 from '../../assets/images/equipments/a8-60.png'
import ATP60E from '../../assets/images/equipments/atp60e.png'
import FTP60E from '../../assets/images/equipments/ftp60e.png'
import FTS60E from '../../assets/images/equipments/fts60e.png'
import Lioness from '../../assets/images/equipments/lioness.png'
import MTS60E from '../../assets/images/equipments/mts60e.png'
import PrimaPremix from '../../assets/images/equipments/prima-premix.png'
import M860 from '../../assets/images/equipments/m8-60.png'

const imagesProduct = {
  A630,
  A860,
  ATP60E,
  FTP60E,
  FTS60E,
  Lioness,
  MTS60E,
  PrimaPremix,
  M860,
}

export {
  imagesProduct,
}
