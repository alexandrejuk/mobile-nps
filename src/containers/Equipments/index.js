import React from 'react'
import PropTypes from 'prop-types'

import CompanyLogo from '../../assets/images/logo.svg'
import { Card } from '../../components'
import styles from './style.module.css'

import A630 from '../../assets/images/equipments/a6-30.png'
import A860 from '../../assets/images/equipments/a8-60.png'
import ATP60E from '../../assets/images/equipments/atp60e.png'
import FTP60E from '../../assets/images/equipments/ftp60e.png'
import FTS60E from '../../assets/images/equipments/fts60e.png'
import Lioness from '../../assets/images/equipments/lioness.png'
import MTS60E from '../../assets/images/equipments/mts60e.png'
import PrimaPremix from '../../assets/images/equipments/prima-premix.png'
import M860 from '../../assets/images/equipments/m8-60.png'

const images = {
  A630,
  A860,
  ATP60E,
  FTP60E,
  FTS60E,
  Lioness,
  MTS60E,
  PrimaPremix,
  M860,
}

const Equipments = ({
  action,
  equipments = [],
}) => {
  return (
    <div className={styles.container}>
      <div className={styles.logo}>
        <img src={CompanyLogo} alt="company logo" />
      </div>
      <div className={styles.equipments}>
        {equipments.map(({
          id,
          image,
          name,
        }) => (
          <div
            key={id}
            className={styles.equipmentItem}
          >
            <Card
              action={() => action(id)}
              image={images[image]}
              title={name}
            />
          </div>
        ))}
      </div>
    </div>
  )
}

Equipments.propTypes = {
  action: PropTypes.func.isRequired,
  equipments: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      image: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
  ).isRequired,
}

export default Equipments
