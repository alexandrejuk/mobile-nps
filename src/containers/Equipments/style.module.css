@import '../../assets/css/colors.css';

.container {
  background-color: var(--color-concrete);
  padding: 10px;
}

.logo {
  margin: auto;
  height: 86px;
  width: 180px;
  padding-bottom: 30px;
}

.logo img {
  height: 100%;
}

.equipmentItem {
  margin: 3px;
}

.equipments {
  display: flex;
  flex-wrap: wrap;
}

@media screen and (max-width: 360px) {
  .equipmentItem {
    margin-bottom: 5px;
  }

}

@media screen and (max-width: 320px) {
  .equipments {
    justify-content: center;
  }
}
