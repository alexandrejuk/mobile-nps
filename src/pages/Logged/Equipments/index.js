import React from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router'
import EquipmentsContainer from '../../../containers/Equipments'
import EquipmentsMock from '../../../utils/api/mocks'

import GAInitialize from '../../../utils/ga'

const Equipments = ({
  history,
}) => {
  GAInitialize('/logged/equipments')
  const goToDetail = id => history.push(`/logged/detail/${id}`)

  return (
    <EquipmentsContainer
      action={goToDetail}
      equipments={EquipmentsMock}
      goBack={history.goBack}
    />
  )
}

Equipments.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
    goBack: PropTypes.func.isRequired,
  }).isRequired,
}

export default withRouter(Equipments)
