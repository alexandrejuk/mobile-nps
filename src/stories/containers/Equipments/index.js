import React from 'react'
import { action } from '@storybook/addon-actions'
import Equipments from '../../../containers/Equipments'

export default {
  title: 'Containers/Equipments',
}

const equipments = [
  {
    id: 1,
    image: 'FTP60E',
    name: 'A8/60',
  },
  {
    id: 2,
    image: 'FTP60E',
    name: 'A8/60',
  },
  {
    id: 3,
    image: 'FTP60E',
    name: 'A8/60',
  },
  {
    id: 4,
    image: 'FTP60E',
    name: 'A8/60',
  },
  {
    id: 5,
    image: 'FTP60E',
    name: 'A8/60',
  },
]

export const Default = () => (
  <Equipments
    goBack={action('go back!')}
    action={action('card action!')}
    equipments={equipments || []}
  />
)
