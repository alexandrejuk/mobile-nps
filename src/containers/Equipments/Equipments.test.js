import React from 'react'
import {
  fireEvent,
  render,
} from '@testing-library/react'

import Equipments from './'

const productsMock = [
  {
    id: 1,
    image: 'M860',
    name: 'A8/60',
  },
  {
    id: 2,
    image: 'M860',
    name: 'A7/30',
  },
  {
    id: 3,
    image: 'M860',
    name: 'A/50',
  },
  {
    id: 4,
    image: 'M860',
    name: 'A1/61',
  },
  {
    id: 5,
    image: 'M860',
    name: 'A8/6',
  },
]

describe('Equipments container', () => {
  let EquipmentsContainer
  it('should render correctly', async () => {
    const callback = jest.fn()
    EquipmentsContainer = render(
      <Equipments
        action={callback}
        equipments={productsMock}
      />
    )
    const companyLogo = await EquipmentsContainer.findByAltText("company logo")
    const imageItems = await EquipmentsContainer.findAllByAltText("item card")
    const imagesSrc = imageItems.map(image => image.getAttribute("src"))

    const headingItems = EquipmentsContainer
      .getAllByRole('heading')
      .map(({ textContent }) => textContent)

    const SampleEquipment = EquipmentsContainer.getByText(productsMock[0].name)

    expect(callback).not.toHaveBeenCalled()
    fireEvent.click(SampleEquipment)
    expect(callback).toHaveBeenCalled()

    expect(companyLogo.getAttribute("src")).toEqual("logo.svg")
    expect(imagesSrc).toEqual(productsMock.map(() => `m8-60.png`))
    expect(headingItems).toEqual(productsMock.map(({ name }) => name))
    expect(EquipmentsContainer.container).toMatchSnapshot()
  })
})
