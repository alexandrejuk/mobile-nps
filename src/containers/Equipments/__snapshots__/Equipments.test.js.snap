// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Equipments container should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="tools"
      role="button"
    >
      <img
        alt="tool help"
        src="help.svg"
      />
    </div>
    <div
      class="logo"
    >
      <img
        alt="company logo"
        src="logo.svg"
      />
    </div>
    <div
      class="equipments"
    >
      <div
        class="equipmentItem"
      >
        <div
          class="card"
          role="button"
          tabindex="0"
        >
          <div
            class="imageContainer"
          >
            <img
              alt="item card"
              src="m8-60.png"
            />
          </div>
          <div
            class="title"
          >
            <h1
              class="primary medium center title bold"
            >
              A8/60
            </h1>
          </div>
        </div>
      </div>
      <div
        class="equipmentItem"
      >
        <div
          class="card"
          role="button"
          tabindex="0"
        >
          <div
            class="imageContainer"
          >
            <img
              alt="item card"
              src="m8-60.png"
            />
          </div>
          <div
            class="title"
          >
            <h1
              class="primary medium center title bold"
            >
              A7/30
            </h1>
          </div>
        </div>
      </div>
      <div
        class="equipmentItem"
      >
        <div
          class="card"
          role="button"
          tabindex="0"
        >
          <div
            class="imageContainer"
          >
            <img
              alt="item card"
              src="m8-60.png"
            />
          </div>
          <div
            class="title"
          >
            <h1
              class="primary medium center title bold"
            >
              A/50
            </h1>
          </div>
        </div>
      </div>
      <div
        class="equipmentItem"
      >
        <div
          class="card"
          role="button"
          tabindex="0"
        >
          <div
            class="imageContainer"
          >
            <img
              alt="item card"
              src="m8-60.png"
            />
          </div>
          <div
            class="title"
          >
            <h1
              class="primary medium center title bold"
            >
              A1/61
            </h1>
          </div>
        </div>
      </div>
      <div
        class="equipmentItem"
      >
        <div
          class="card"
          role="button"
          tabindex="0"
        >
          <div
            class="imageContainer"
          >
            <img
              alt="item card"
              src="m8-60.png"
            />
          </div>
          <div
            class="title"
          >
            <h1
              class="primary medium center title bold"
            >
              A8/6
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
